pycape/function_ref.py

```diff
@@ -84,28 +84,57 @@ def from_json(cls, token_path: Union[str, os.PathLike]):
             ValueError: if the json token file doesn't exist or, the token file
                 doesn't contain a `function_id` or a `function_token`.
         """
-        if isinstance(token_path, str):
-            token_path = pathlib.Path(token_path)
 
-        if token_path.exists():
-            with open(token_path, "r") as f:
-                token_config = json.load(f)
-        else:
-            raise ValueError(
-                "Couldn't find the token json file with the provided "
-                f"path: {str(token_path)}"
-            )
-
-        function_id = token_config.get("function_id")
+        try:
+            if isinstance(function_json, pathlib.Path):
+                function_config = json.loads(str(function_json))
+            else:
+                function_config = json.loads(function_json)
+        except json.JSONDecodeError:
+            if isinstance(function_json, str):
+                function_json = pathlib.Path(function_json)
+
+            if function_json.exists():
+                with open(function_json, "r") as f:
+                    function_config = json.load(f)
+            else:
+                raise ValueError(
+                    "Couldn't parse the json string or couldn't find the "
+                    f"function json file with the provided path: {str(function_json)}"
+                )
+
+        function_id = function_config.get("function_id")
         if function_id is None:
             raise ValueError("Couldn't find a `function_id` in the token file provided")
 
-        function_token = token_config.get("function_token")
+        function_token = function_config.get("function_token")
         if function_token is None:
             raise ValueError(
                 "Couldn't find a `function_token` in the token file provided"
             )
 
-        function_checksum = token_config.get("function_checksum")
+        function_checksum = function_config.get("function_checksum")
 
         return cls(function_id, function_token, function_checksum)
+
+    def to_json(self, path: Optional[Union[str, os.PathLike]] = None):
+        """
+        Save function ID, token & checksum in a json file or as a json string
+        if path is None.
+
+        Args:
+            path: optional file path to save function ID, token & checksum
+            as a json string. Otherwise it will return a json string.
+        """
```

Comment:
this docstring needs to be reformatted properly. extra lines for each arg need to be indented since the `path` arg is not being rendered properly on the [docs site](https://deploy-preview-101--vocal-profiterole-b755d1.netlify.app/pycape.function_ref.html#pycape.function_ref.FunctionRef). also you will need to add a `Returns:` block that describes the output type. `from_json` above needs the same treatment